runtime/westend/src/weights/xcm/mod.rs

```diff
@@ -0,0 +1,191 @@
+					AssetTypes::Unknown => Weight::MAX,
+				})
+				.fold(0, |acc, x| acc.saturating_add(x)),
+			_ => Weight::MAX,
+		}
+	}
+}
+
+impl WeighMultiAssets for MultiAssets {
+	fn weigh_multi_assets(&self, balances_weight: Weight) -> Weight {
+		self.inner()
+			.into_iter()
+			.map(|m| <AssetTypes as From<&MultiAsset>>::from(m))
+			.map(|t| match t {
+				AssetTypes::Balances => balances_weight,
+				AssetTypes::Unknown => Weight::MAX,
+			})
+			.fold(0, |acc, x| acc.saturating_add(x))
+	}
+}
+
+pub struct WestendXcmWeight<Call>(core::marker::PhantomData<Call>);
+impl<Call> XcmWeightInfo<Call> for WestendXcmWeight<Call> {
+	fn withdraw_asset(assets: &MultiAssets) -> Weight {
+		assets.weigh_multi_assets(XcmBalancesWeight::<Runtime>::withdraw_asset())
+	}
+	fn reserve_asset_deposited(assets: &MultiAssets) -> Weight {
+		assets.weigh_multi_assets(XcmGeneric::<Runtime>::reserve_asset_deposited())
+	}
+	// TODO none of these need effects
+	fn receive_teleported_asset(assets: &MultiAssets) -> Weight {
+		assets.weigh_multi_assets(XcmBalancesWeight::<Runtime>::receive_teleported_asset())
+	}
+	fn query_response(_query_id: &u64, _response: &Response, max_weight: &u64) -> Weight {
+		*max_weight
+	}
+	fn transfer_asset(assets: &MultiAssets, _dest: &MultiLocation) -> Weight {
+		assets.weigh_multi_assets(XcmBalancesWeight::<Runtime>::transfer_asset())
+	}
+	fn transfer_reserve_asset(
+		assets: &MultiAssets,
+		_dest: &MultiLocation,
+		_xcm: &Xcm<()>,
+	) -> Weight {
+		assets.weigh_multi_assets(XcmBalancesWeight::<Runtime>::transfer_reserve_asset())
+	}
+	fn transact(
+		_origin_type: &OriginKind,
+		_require_weight_at_most: &u64,
+		_call: &DoubleEncoded<Call>,
+	) -> Weight {
+		XcmGeneric::<Runtime>::transact()
+	}
+	fn hrmp_new_channel_open_request(
+		_sender: &u32,
+		_max_message_size: &u32,
+		_max_capacity: &u32,
+	) -> Weight {
+		XcmGeneric::<Runtime>::hrmp_new_channel_open_request()
+	}
+	fn hrmp_channel_accepted(_recipient: &u32) -> Weight {
+		XcmGeneric::<Runtime>::hrmp_channel_accepted()
+	}
+	fn hrmp_channel_closing(_initiator: &u32, _sender: &u32, _recipient: &u32) -> Weight {
+		XcmGeneric::<Runtime>::hrmp_channel_closing()
+	}
+	fn clear_origin() -> Weight {
+		XcmGeneric::<Runtime>::clear_origin()
+	}
+	fn descend_origin(who: &InteriorMultiLocation) -> Weight {
+		XcmGeneric::<Runtime>::descend_origin(who)
+	}
+	fn report_error(
+		_query_id: &QueryId,
+		_dest: &MultiLocation,
+		max_response_weight: &u64,
+	) -> Weight {
+		*max_response_weight
+	}
+	fn relayed_from(_who: &Junctions, _message: &Box<Xcm<Call>>) -> Weight {
+		XcmGeneric::<Runtime>::relayed_from()
+	}
+
+	fn deposit_asset(
+		assets: &MultiAssetFilter,
+		_max_assets: &u32, // TODO use max assets?
+		_dest: &MultiLocation,
+	) -> Weight {
+		assets.weigh_multi_assets(XcmBalancesWeight::<Runtime>::deposit_asset())
+	}
+	fn deposit_reserve_asset(
+		assets: &MultiAssetFilter,
+		_max_assets: &u32, // TODO use max assets?
+		_dest: &MultiLocation,
+		_xcm: &Xcm<()>,
```

Comment:
these `xcm` are being forwarded, and should be taken into account by multiplying them with something like `generic_xcm_weights::send_xcm()`. A few other similar instances exist.

Reply:
UPDATE: no `Vec<_>` involved, at most one xcm forwarding, but my comment over all still holds.